src/script_interface/get_value.hpp

```diff
@@ -256,6 +256,21 @@ template <typename T>
 void set_from_args(T &dst, VariantMap const &vals, const char *name) {
   dst = get_value<T>(vals, name);
 }
+
+/**
+ * @brief Convert a list of 2-tuples to an unordered map.
+ * Typically used for python dict objects, which have to be passed as
+ * lists of tuples using <tt>list(argument.items())</tt>.
+ */
+template <typename K, typename T>
+std::unordered_map<K, T> get_map(const std::vector<Variant> &vv) {
+  std::unordered_map<K, T> ret{};
+  for (auto const &kv : vv) {
+    auto const kv_vec = get_value<std::vector<Variant>>(kv);
```

Comment:
i'm not sure if i understand this correctly... is the input a vector of a variant and a possible type of this variant is a vector of a variant?

Reply:
We use this `get_map()` function when we know the Variant object is packing a python `dict` which has been converted to a `list`. Each item in the `std::vector<Variant>` is itself a `std::vector<Variant>` of size 2, the first item is of type `K` (typically  `int`) and the second item is of type `T`.

I've tried to update the `Variant` object to pack a `std::unordered_map<int, Variant>` and avoid using this construction, but a template error message resisted for 1 hour...

Reply:
ok I've finally got the `dict <-> std::unordered_map<int, Variant>` conversion to work, the code above can be removed if we change the API to pass a `dict` to coupling-based fields instead of a flattened `dict_items` object of the form `[(pid1, scale1), (pid2, scale2), ...]`.